var/spack/repos/builtin/packages/aomp/package.py

```diff
@@ -7,113 +7,282 @@
+                sha256=aomp35['llvm'],
+                expand=True,
+                destination='aomp-dir',
+                placement='amd-llvm-project',
+                when='@3.5.0')
+
+            resource(
+                name='flang',
+                url=tools_url + '/flang/archive/rocm-3.5.0.tar.gz',
+                sha256=aomp35['flang'],
+                expand=True,
+                destination='aomp-dir',
+                placement='flang',
+                when='@3.5.0')
+
+            resource(
+                name='aomp-extras',
+                url=tools_url + '/aomp-extras/archive/rocm-3.5.0.tar.gz',
+                sha256=aomp35['extras'],
+                expand=True,
+                destination='aomp-dir',
+                placement='aomp-extras',
+                when='@3.5.0')
+
+            resource(
+                name='hip-on-vdi',
+                url=tools_url + '/hip/archive/aomp-3.5.0.tar.gz',
+                sha256=aomp35['hip'],
+                expand=True,
+                destination='aomp-dir',
+                placement='hip-on-vdi',
+                when='@3.5.0')
+
+            resource(
+                name='vdi',
+                url=tools_url + '/rocclr/archive/aomp-3.5.0.tar.gz',
+                sha256=aomp35['vdi'],
+                expand=True,
+                destination='aomp-dir',
+                placement='vdi',
+                when='@3.5.0')
+
+            resource(
+                name='opencl-on-vdi',
+                sha256=aomp35['opencl'],
+                url=compute_url +
+                '/ROCm-OpenCL-Runtime/archive/aomp-3.5.0.tar.gz',
+                expand=True,
+                destination='aomp-dir',
+                placement='opencl-on-vdi',
+                when='@3.5.0')
+    # Revert back to .amdgcn.bc naming scheme for 3.8.0
+    patch('0001-Add-amdgcn-to-devicelibs-bitcode-names-3.8.patch',
+          working_dir='aomp-dir/amd-llvm-project', when='@3.8.0')
+
+    # Revert back to .amdgcn.bc naming scheme for 3.7.0
+    patch('0001-Add-amdgcn-to-devicelibs-bitcode-names.patch',
+          working_dir='aomp-dir/amd-llvm-project', when='@3.7.0')
 
     def patch(self):
-        kwargs = {'ignore_absent': False, 'backup': False, 'string': False}
-        with working_dir('aomp-dir/hip-on-vdi'):
-            match = '^#!/usr/bin/python'
-            python = self.spec['python'].command
-            substitute = "#!{python}".format(python=python)
-            files = [
-                'hip_prof_gen.py', 'vdi/hip_prof_gen.py'
-            ]
-            filter_file(match, substitute, *files, **kwargs)
+        # Make sure python2.7 is used for the generation of hip header
+        if self.spec.version == Version('3.5.0'):
+            kwargs = {'ignore_absent': False, 'backup': False, 'string': False}
+            with working_dir('aomp-dir/hip-on-vdi'):
+                match = '^#!/usr/bin/python'
+                python = self.spec['python'].command
```

Comment:
```suggestion
                python = self.spec['python'].command.path
```
The former might work, but this is more clear